src/handler/mod.rs

```diff
@@ -351,7 +363,7 @@ impl Handler {
             .filter(|id| self.browser_contexts.contains(id))
             .unwrap_or_else(|| self.default_browser_context.clone());
         let target = Target::new(
-            event.target_info,
+            event.target_info.clone(),
```

Comment:
This is one of the problems I talked about in the issue. Now that we want to consume events in the target and in the events there is a lot of cloning involved. 
I think it would be more beneficial, at least for now, if we send the event first via the event stream and clone the there and consume the event internally afterwards.